layers/+distributions/spacemacs-base/funcs.el

```diff
@@ -380,16 +380,22 @@ projectile cache when it's possible.
 When ASK-USER is non-nil, user will be asked to confirm file
 removal."
   (interactive "f")
-  (when (and filename (file-exists-p filename))
-    (let ((buffer (find-buffer-visiting filename)))
-      (when buffer
-        (kill-buffer buffer)))
-    (when (or (not ask-user)
-              (yes-or-no-p "Are you sure you want to delete this file? "))
-      (delete-file filename)
+  (let ((is-dir (file-directory-p filename)))
+    (when (and (file-exists-p filename)
+               (or (not ask-user)
+                   (yes-or-no-p
+                    (message  "Are you sure you want to delete this %s? "
+                              (if is-dir "directory" "file")))))
+      (spacemacs/kill-file-or-dir-buffers filename)
+      (if (file-directory-p filename)
+          (delete-directory filename t)
```

Comment:
Well, we delete the files either way. The question is whether to delete them permanently (as `rm` does) or put them in the trash can (as file managers do). If we move the files to the trash then the user can still restore them if he/she realizes it was a mistake to delete the files. I think that should be the default, and that's also the default behavior of `delete-directory` when called interactively (but only when called interactively).

Same comment goes for `delete-file` below, I just missed it before.